geode-core/src/distributedTest/java/org/apache/geode/internal/cache/partitioned/fixed/FixedPartitioningWithTransactionDistributedTest.java

```diff
@@ -0,0 +1,375 @@
+        .setPartitionAttributes(factory.create()).create(regionName);
+    CacheServer server = cacheRule.getCache().addCacheServer();
+    server.setPort(0);
+    server.start();
+    return server.getPort();
+  }
+
+  private void createClientRegion(boolean connectToFirstPort, int... ports) {
+    clientCacheRule.createClientCache();
+
+    CacheServerTestUtil.disableShufflingOfEndpoints();
+    PoolImpl pool;
+    try {
+      pool = getPool(ports);
+    } finally {
+      CacheServerTestUtil.enableShufflingOfEndpoints();
+    }
+
+    ClientRegionFactory crf =
+        clientCacheRule.getClientCache().createClientRegionFactory(ClientRegionShortcut.LOCAL);
+    crf.setPoolName(pool.getName());
+    crf.create(regionName);
+
+    if (ports.length > 1 && connectToFirstPort) {
+      // first connection to the first port in the list
+      pool.acquireConnection(new ServerLocation(hostName, ports[0]));
+    }
+  }
+
+  private PoolImpl getPool(int... ports) {
+    PoolFactory factory = PoolManager.createFactory();
+    for (int port : ports) {
+      factory.addServer(hostName, port);
+    }
+    factory.setPRSingleHopEnabled(false);
+    return (PoolImpl) factory.create(uniqueName);
+  }
+
+  private void registerFunctions() {
+    FunctionService.registerFunction(new MySuspendTransactionFunction());
+    FunctionService.registerFunction(new MyResumeTransactionFunction());
+  }
+
+  private TransactionId doFunctionTransactionAndSuspend() {
+    Region region = cacheRule.getCache().getRegion(regionName);
+    TXManagerImpl manager = cacheRule.getCache().getTxManager();
+    return doFunctionTransactionAndSuspend(region, manager);
+  }
+
+  private TransactionId doFunctionTransactionAndSuspend(Region region,
+      CacheTransactionManager manager) {
+    Execution execution = FunctionService.onRegion(region);
+    manager.begin();
+    Set keySet = new HashSet();
+    keySet.add(2);
+    ResultCollector resultCollector = execution.execute(new MySuspendTransactionFunction());
+    resultCollector.getResult();
+    return manager.suspend();
+  }
+
+  private void resumeFunctionTransaction(TransactionId transactionId) {
+    Region region = cacheRule.getCache().getRegion(regionName);
+    TXManagerImpl manager = cacheRule.getCache().getTxManager();
+    resumeFunctionTransaction(transactionId, region, manager);
+  }
+
+  private void resumeFunctionTransaction(TransactionId transactionId, Region region,
+      CacheTransactionManager manager) {
+    Execution execution = FunctionService.onRegion(region);
+    manager.resume(transactionId);
+    try {
+      Set keySet = new HashSet();
+      keySet.add(3);
+      ResultCollector resultCollector =
+          execution.withFilter(keySet).execute(new MyResumeTransactionFunction());
+      resultCollector.getResult();
+    } finally {
+      manager.rollback();
+    }
+  }
+
+  public static class MyFixedPartitionResolver implements FixedPartitionResolver {
```

Comment:
I would change these inner classes to private if you can.

Reply:
Have to change private to public due to following exception
Caused by: java.lang.NoSuchMethodException: org.apache.geode.internal.cache.partitioned.fixed.FixedPartitioningWithTransactionDistributedTest$MySuspendTransactionFunction